### PR TITLE
Migrate exceptions to PHP 8 syntax

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 1]
```diff
@@ -768,46 +488,25 @@ public static function invalidFetchMode(string $className, string $fetchMode): s
...
-     * @return MappingException
-     */
-    public static function invalidCascadeOption(array $cascades, $className, $propertyName)
+    /** @param string[] $cascades */
+    public static function invalidCascadeOption(array $cascades, string $className, string $propertyName): self
     {
-        $cascades = implode(', ', array_map(static function ($e) {
-            return "'" . $e . "'";
-        }, $cascades));
+        $cascades = implode(', ', array_map(static fn ($e) => "'" . $e . "'", $cascades));
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/MappingException.php]:
```suggestion
        $cascades = implode(', ', array_map(static fn (string $e) => "'" . $e . "'", $cascades));
```

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 2]
```diff
@@ -865,8 +547,7 @@ public static function illegalOverrideOfInheritedProperty($className, $propertyN
         );
     }
 
-    /** @return self */
-    public static function invalidIndexConfiguration($className, $indexName)
+    public static function invalidIndexConfiguration($className, $indexName): self
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/MappingException.php]: Types?

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 3]
```diff
@@ -877,8 +558,7 @@ public static function invalidIndexConfiguration($className, $indexName)
         );
     }
 
-    /** @return self */
-    public static function invalidUniqueConstraintConfiguration($className, $indexName)
+    public static function invalidUniqueConstraintConfiguration($className, $indexName): self
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/MappingException.php]: Types?

**Review by derrabus — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 4]
```diff
@@ -329,12 +220,8 @@ public static function joinColumnMustPointToMappedField($className, $joinColumn)
...
     }
 
-    /**
-     * @param class-string $className
-     *
-     * @return MappingException
-     */
-    public static function classIsNotAValidEntityOrMappedSuperClass($className)
+    /** @param class-string $className */
+    public static function classIsNotAValidEntityOrMappedSuperClass($className): self
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/MappingException.php]:
```suggestion
    public static function classIsNotAValidEntityOrMappedSuperClass(string $className): self
```

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 5]
```diff
@@ -448,12 +298,8 @@ public static function unsupportedOptimisticLockingType($entity, $fieldName, $un
...
     }
 
-    /**
-     * @param string|null $path
-     *
-     * @return MappingException
-     */
-    public static function fileMappingDriversRequireConfiguredDirectoryPath($path = null)
+    /** @param string|null $path */
+    public static function fileMappingDriversRequireConfiguredDirectoryPath($path = null): self
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/MappingException.php]:
```suggestion
    public static function fileMappingDriversRequireConfiguredDirectoryPath(string|null $path = null): self
```

File: lib/Doctrine/ORM/Mapping/MappingException.php [hunk 6]
```diff
@@ -768,46 +488,25 @@ public static function invalidFetchMode(string $className, string $fetchMode): s
...
-     * @return MappingException
-     */
-    public static function invalidCascadeOption(array $cascades, $className, $propertyName)
+    /** @param string[] $cascades */
+    public static function invalidCascadeOption(array $cascades, string $className, string $propertyName): self
     {
-        $cascades = implode(', ', array_map(static function ($e) {
-            return "'" . $e . "'";
-        }, $cascades));
+        $cascades = implode(', ', array_map(static fn (string $e) => "'" . $e . "'", $cascades));
```
Comment (SenseException) [inline lib/Doctrine/ORM/Mapping/MappingException.php]:
```suggestion
        $cascades = implode(', ', array_map(static fn (string $e): string => "'" . $e . "'", $cascades));
```
... I guess

**Review by SenseException — DISMISSED** [review]

**Review by derrabus — APPROVED** [review]

**Pull request merged**